### PR TITLE
add support for binary classifiers edge case

### PR DESCRIPTION
**Pull request by aditya1503**
## Summary

> 🎯 **Purpose**: Add support for binary classifiers with near 0.5 probability

### CODE REVIEW

File: cleanlab/filter.py [hunk 1]
```diff
@@ -446,7 +446,11 @@ class 0, 1, ..., K-1.
     if filter_by not in ["low_self_confidence", "low_normalized_margin"]:
         # Remove label issues if given label == model prediction if issues haven't been removed yet
         pred = pred_probs.argmax(axis=1)
-        label_issues_mask[pred == labels] = False
+        mask = pred == labels
+        if K == 2:
+            # Set mask to False wherever pred_probs == 0.5 where num_classes == 2
+            mask = mask & (pred_probs[:, 0] != 0.5)
```
Comment (aditya1503) [inline cleanlab/filter.py:452]: Should this be between +/- (0.5 + TINY_VALUE)?

File: cleanlab/filter.py [hunk 2]
```diff
@@ -446,7 +446,11 @@ class 0, 1, ..., K-1.
...
+        label_issues_mask[mask] = False
```
Comment (jwmueller) [inline cleanlab/filter.py:453]:
```suggestion
        pred_confidence = pred_probs[:, pred]  # TODO: fix this with the correct syntax
        mask = mask & (pred_confidence < 1/K + EPS)
        label_issues_mask[mask] = False
```

Comment (jwmueller) [reply cleanlab/filter.py:453]: @aditya1503 what about doing it like this ^ instead so it's more general solution?

File: cleanlab/filter.py [hunk 3]
```diff
@@ -446,7 +446,11 @@ class 0, 1, ..., K-1.
...
```
Comment (jwmueller) [inline cleanlab/filter.py:454]: Don't forget any updates to find_label_issues from now on must also be considered for find_label_issues_batched

File: cleanlab/filter.py (hunk 1 above)
Comment (jwmueller) [reply cleanlab/filter.py:452]: yes it should be:

pred_probs[:,given_label] > 0.5 - EPS

File: cleanlab/filter.py (hunk 3 above)
Comment (jwmueller) [reply cleanlab/filter.py:454]: Note this actually also happening in this method: https://github.com/cleanlab/cleanlab/blob/master/cleanlab/count.py#L153

I'm worried it is too unsafe to just do this update here, we are likely to miss other related methods where it needs to happen too

File: tests/test_multilabel_classification.py [hunk 4]
```diff
@@ -17,23 +17,24 @@
...
+from cleanlab.internal import multilabel_scorer as ml_scorer
+from cleanlab.internal.multilabel_utils import get_onehot_num_classes, onehot2int, stack_complement
+from cleanlab.multilabel_classification import filter
 from cleanlab.multilabel_classification.dataset import (
     common_multilabel_issues,
-    rank_classes_by_multilabel_quality,
-    overall_multilabel_health_score,
     multilabel_health_summary,
+    overall_multilabel_health_score,
+    rank_classes_by_multilabel_quality,
```
Comment (aditya1503) [inline tests/test_multilabel_classification.py:35]: These file changes are just iSort, no code changes here

File: cleanlab/filter.py (hunk 3 above)
Comment (aditya1503) [reply cleanlab/filter.py:450]: I've added support in all places I could find

File: cleanlab/count.py [hunk 5]
```diff
@@ -224,6 +228,15 @@ def _num_label_issues_multilabel(
     return sum(issues_idx)
 
 
+def _reduce_issues(pred_probs, labels, num_classes):
+    pred = pred_probs.argmax(axis=1)
+    mask = pred == labels
+    if num_classes == 2:
+        # Set mask to False wherever pred_probs == 0.5 where num_classes == 2
+        mask = mask | ((pred_probs[:, 0] < 0.5 - EPSILON) | (pred_probs[:, 0] > 0.5 + EPSILON))
```
Comment (jwmueller) [inline cleanlab/count.py:236]: I think what I suggested would be cleaner code:

mask = pred_probs[:,given_label] > 0.5 - EPS

that in future could generalize to more classes beyond K = 2

Comment (jwmueller) [reply cleanlab/count.py:236]: even cleaner could be:

```
pred_probs[:,given_label] += EPS
pred = pred_probs.argmax(axis=1)
mask = pred == labels
```

**Pull request closed without merging**